pensieve/tests/test_config.py

```diff
@@ -38,39 +39,53 @@ def test_template_expansion(self, experiments):
         )
         spec = config.AnalysisSpec.from_dict(toml.loads(config_str))
         cfg = spec.resolve(experiments[0])
-        metric = [m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.name == "my_cool_metric"][0]
+        metric = [m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.metric.name == "my_cool_metric"][
+            0
+        ].metric
         assert "agg_histogram_mean" not in metric.select_expr
         assert "json_extract_histogram" in metric.select_expr
 
     def test_recognizes_metrics(self, experiments):
         config_str = dedent(
             """
             [metrics]
-            weekly = ["view_about_logins"]
+            weekly = [{metric = "view_about_logins", treatment = "bootstrap_mean"}]
             """
         )
         spec = config.AnalysisSpec.from_dict(toml.loads(config_str))
         cfg = spec.resolve(experiments[0])
         assert (
-            len([m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.name == "view_about_logins"]) == 1
+            len(
+                [
+                    m
+                    for m in cfg.metrics[AnalysisPeriod.WEEK]
+                    if m.metric.name == "view_about_logins"
+                ]
+            )
+            == 1
         )
 
     def test_duplicate_metrics_are_okay(self, experiments):
         config_str = dedent(
             """
             [metrics]
-            weekly = ["unenroll", "unenroll", "active_hours"]
+            weekly = [
+                {metric = "unenroll", treatment = "bootstrap_mean"},
+                {metric = "unenroll", treatment = "bootstrap_mean"},
+                {metric = "active_hours", treatment = "bootstrap_mean"}]
             """
         )
         spec = config.AnalysisSpec.from_dict(toml.loads(config_str))
         cfg = spec.resolve(experiments[0])
-        assert len([m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.name == "unenroll"]) == 1
+        assert (
+            len([m for m in cfg.metrics[AnalysisPeriod.WEEK] if m.metric.name == "unenroll"]) == 1
+        )
 
     def test_data_source_definition(self, experiments):
         config_str = dedent(
             """
             [metrics]
-            weekly = ["spam"]
+            weekly = [{metric = "spam", treatment = "bootstrap_mean"}]
```

Comment:
I'm open to alternative representations for combining a metric with a statistical treatment. But this at least seems alright to me

Reply:
elements of a toml 0.5 list have to be homogenous so we can't mix strings and objects, which is the only reason i don't love this representation. "You can only define statistics for custom metrics" sounds kind of okay to me; we could make it a property of the metric specification.

Reply:
Are there cases where only metrics without any treatment should be calculated?

Reply:
No, but I think that I think the builtins should have defaults and a fallback.